### DISCUSSION TITLE
Supporting for Heterogeneous temporal graph (dynamic heterogeneous graph)

### DISCUSSION THREAD

**Discussion by Julian-Yu** [pyg-team/pytorch_geometric#8101] [Ideas]
## Supporting for Heterogeneous temporal graph (dynamic heterogeneous graph)
I see that Heterogeneous Graph Neural Networks and Temporal Graph Neural Networks are already supported in PyG, and there are data structures HeteroData and TemporalData for storing them, but there is no support for Heterogeneous Temporal Graphs (Dynamic Heterogeneous Graphs) so far, especially for the data structures. I would appreciate it if you could support the above data structures as soon as possible.

**Comment by rusty1s** [comment]
This is currently not planned yet (at least not for short-term). We would first need to ramp up `TemporalData` and temporal GNN support in PyG before we integrate support for heterogeneous graphs. Note that most applications for `TemporalData` already can hold heterogeneous graphs though by just stacking nodes of different types in the node dimension.
